src/Symfony/Component/Messenger/Handler/Locator/AbstractHandlerLocator.php

```diff
@@ -11,36 +11,41 @@
 
 namespace Symfony\Component\Messenger\Handler\Locator;
 
+use Symfony\Component\Messenger\Envelope;
 use Symfony\Component\Messenger\Exception\NoHandlerForMessageException;
 
 /**
  * @author Miha Vrhovnik <miha.vrhovnik@gmail.com>
  * @author Samuel Roze <samuel.roze@gmail.com>
+ *
+ * @internal
  */
 abstract class AbstractHandlerLocator implements HandlerLocatorInterface
 {
-    public function resolve($message): callable
+    public function getHandler(Envelope $envelope, bool $allowNoHandler = false): ?callable
```

Comment:
`$allowNoHandler` is not used for now, but I'd like we add it right now in the interface and replace  `AllowNoHandlerMiddleware` later on (using exceptions as a signaling mean is making the "no-handler" path slow for no real reasons.)

Reply:
(agreed, removed)